src/components/ManageAssets/ManageAssets.tsx

```diff
@@ -0,0 +1,161 @@
+    };
+
+    const handleNoAccess = () => {
+      alert("User has no manage asset privileges.");
+    };
+
+    const handleConnectMetamask = async (event: { preventDefault: () => void }) => {
+      event.preventDefault();
+      await connectToMetamask();
+      dispatch(loadAdminAddress());
+    };
+
+    return (
+      <Dropdown className={`${className}`} alignRight>
+        <Dropdown.Toggle
+          variant="primary"
+          id="dropdown-basic"
+          {...(userNotLoginMetamask && { onClick: handleConnectMetamask })}
+          {...(userNoManagePrivileges && { onClick: handleNoAccess })}
+        >
+          Manage Assets
+        </Dropdown.Toggle>
+
+        <Dropdown.Menu>
+          {userWalletId === holderAddress && (
+            <>
+              <Dropdown.Item onClick={handleSelectedTransfer}>Transfer holdership</Dropdown.Item>
+              {isApprovedAddressPending && <Dropdown.Item>Confirm endorse change of beneficiary</Dropdown.Item>}
+            </>
+          )}
+          {userWalletId === beneficiaryAddress && (
+            <>
+              {!isApprovedAddressPending && (
+                <Dropdown.Item onClick={handleSelectedTransfer}>Endorse change of beneficiary</Dropdown.Item>
+              )}
+            </>
+          )}
+          {userWalletId === holderAddress && userWalletId === beneficiaryAddress && (
+            <Dropdown.Item onClick={handleSelectedTransfer}>Surrender document</Dropdown.Item>
+          )}
+        </Dropdown.Menu>
+      </Dropdown>
+    );
+  }
+)`
+  &.show {
+    > .dropdown-toggle {
+      color: ${vars.white};
+    }
+  }
+
+  .btn-primary {
+    ${mixin.fontSourcesansproBold}
+    color: ${vars.white};
+
+    &:not(:disabled):not(.disabled):active,
+    &:focus,
+    &.focus,
+    &:hover,
+    &:active,
+    &.active {
+      color: ${vars.white};
+    }
+  }
+
+  .dropdown-toggle {
+    &::after {
+      margin-left: 10px;
+    }
+  }
+
+  .dropdown-item {
+    padding: 8px 15px;
+  }
+`;
+
+interface ManageAssetsProps {
+  document: any;
+}
+
+export const ManageAssets = ({ document }: ManageAssetsProps) => {
+  const {
+    userWalletId,
+    approvedBeneficiaryAddress,
+    approvedHolderAddress,
+    beneficiaryAddress,
+    holderAddress,
+    metamaskAccountError,
+  } = useSelector((state: any) => ({
+    userWalletId: state.admin.adminAddress,
+    approvedBeneficiaryAddress: state.token.approvedBeneficiaryAddress,
+    approvedHolderAddress: state.token.approvedHolderAddress,
+    beneficiaryAddress: state.token.beneficiaryAddress,
+    holderAddress: state.token.holderAddress,
+    metamaskAccountError: state.admin.metamaskAccountError,
+  }));
+
+  return (
+    <div className="py-4">
+      <div className="container-custom">
+        <div className="row no-gutters align-items-center">
+          <div className="col-12 col-md-auto">
+            <AssetInfo document={document} />
+          </div>
+          <FeatureFlag name="TITLE_TRANSFER">
```

Comment:
no callback feature ?

Reply:
not necessary?

Reply:
by default I can see the manage asset button. It should be hidden untill the feature is developed completely and old manage asset link should be shown.